modules/core/arrow-test/src/main/kotlin/arrow/test/laws/BicrosswalkLaws.kt

```diff
@@ -0,0 +1,76 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.Kind2
+import arrow.core.ForListK
+import arrow.core.ListK
+import arrow.core.extensions.listk.align.align
+import arrow.core.extensions.listk.eqK.eqK
+import arrow.core.k
+import arrow.typeclasses.Align
+import arrow.typeclasses.Bicrosswalk
+import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import kotlin.math.abs
+
+object BicrosswalkLaws {
+
+  fun <T> laws(
+    BCW: Bicrosswalk<T>,
+    gen: Gen<Kind2<T, Int, Int>>,
+    EQ: Eq<Kind2<T, String, String>>
+  ): List<Law> {
+
+    val funGen = object : Gen<(Int) -> Kind<ForListK, String>> {
+      override fun constants(): Iterable<(Int) -> ListK<String>> = listOf(
+        { _: Int -> ListK.empty<String>() }, { _: Int -> ListK.just("value") }
+      )
+
+      override fun random(): Sequence<(Int) -> ListK<String>> = generateSequence({ int: Int -> List(abs(int % 1000)) { "$it" }.k() }) { it }
+    }
```

Comment:
I think there are primitive function generators in our tests already :) Although those are literally worthless imo, as the author of a property testing library I might be biased tho :)

Reply:
what lib are your the author of?

Reply:
https://github.com/1Jajen1/propCheck
It's undergoing quite the rewrite tho. (the pr for 1.0)The current one is good, but the rewrite aims to make it easier to use along with giving it more power. This is arguably better at property based testing than kotlin test if you only consider property testing. I plan on having the rewrite done by 2020 with better docs etc. But that may be delayed depending on work and/university :/

Reply:
looks interesting. i don't have much experience yet with property based testing. 
do you plan to introduce propCheck to the arrow codebase?

Reply:
I'd love to, but don't have the time for it atm. Raul and I were talking about this already half a year ago, but I had an annoying hospital stay + time to recover and catch up with work and university made that impossible.

Maybe another attempt next year when the rewrite is done, but that also depends on what the other maintainers think.

Reply:
sorry to hear that about your hospital stay. i hope you are fully recovered! i will definitely check out propCheck.

Reply:
Yeah all good now :)
Property based testing is **amazing** if you can get it right, some really really helpful resources: https://williamyaoh.com/posts/2019-11-24-design-and-testing-articles.html
Everything related to property based testing on the testing part is great, especially the ones by Oskar Wickström